src/types/bet_note.nr

```diff
@@ -0,0 +1,122 @@
+use dep::aztec::{
+    note::{
+        note_header::NoteHeader,
+        note_interface::NoteInterface,
+        utils::compute_note_hash_for_read_or_nullify,
+    },
+    oracle::{
+        rand::rand,
+        get_secret_key::get_secret_key,
+        get_public_key::get_public_key
+    },
+
+    log::emit_encrypted_log,
+    context::PrivateContext,
+    types::{
+        address::AztecAddress,
+    }
+};
+
+struct BetNote {
+    owner: AztecAddress,
+    randomness: Field,
+    bet: bool, // Heads is 0 (False). Tails is 1 (True).
+    header: NoteHeader
+}
+
+global BET_NOTE_LEN: Field = 3;
+
+impl BetNote {
+    pub fn new(_owner: AztecAddress, _bet: bool) -> Self {
+        BetNote {
+            owner: _owner,
+            randomness: rand(),
+            bet: _bet,
+            header: NoteHeader::empty(),
+        }
+    }
+
+    fn serialize(self) -> [Field; BET_NOTE_LEN] {
+        [self.owner.address, self.randomness, self.bet as Field]
+    }
+
+    fn deserialize(preimage: [Field; BET_NOTE_LEN]) -> Self {
+        BetNote {
+            owner: AztecAddress::new(preimage[0]),
+            randomness: preimage[1],
+            bet: preimage[2] as bool,
+            header: NoteHeader::empty(),
+        }
+    }
+
+    fn compute_note_hash(self) -> Field {
+        dep::std::hash::pedersen_hash([
+            self.owner.address,
+            self.randomness,
+            self.bet as Field
+        ])
+    }
+
+    fn compute_nullifier(self) -> Field {
+        let note_hash_for_nullify = compute_note_hash_for_read_or_nullify(BetNoteMethods, self);
+        let secret = get_secret_key(self.owner.address);
+        dep::std::hash::pedersen_hash([
+            note_hash_for_nullify,
+            secret.low,
+            secret.high,
+        ])
+    }
+
+    fn set_header(&mut self, header: NoteHeader) {
+        self.header = header;
+    }
+
+    fn broadcast(self, context: &mut PrivateContext, slot: Field) {
+        let encryption_pub_key = get_public_key(self.owner.address);
```

Comment:
This will be the function that will broadcast this note to the user. When creating the bet note we will need to also broadcast it for the house (same note)